### ISSUE TITLE
Wrong block hash?

### ISSUE THREAD

**Issue by biofreeze** [dcSpark/carp#176]
Not sure what's going on here, but i'm grabbing the last block from the db like so:
```
carp_mainnet=# SELECT * FROM "Block" ORDER BY "Block".id DESC LIMIT 1;
-[ RECORD 1 ]---------------------------------------------------------------
id      | 169013
era     | 0
hash    | \x869b30014807b9b158c8d24fb406deea93f39893169dc47c32a2cb6e6f642e2d
height  | 169012
epoch   | 7
slot    | 169043
payload | \x
```

As you can see here, it says the block hash is `869b30014807b9b158c8d24fb406deea93f39893169dc47c32a2cb6e6f642e2d`. I looked up block `169012` on CardanoScan, but this is not the correct hash. The slot is however correct (169043).

Link for the block here on CardanoScan https://cardanoscan.io/block/169012

When I try to start Carp it gives the following error:
```
2024-03-03 13:27:18 Applying all pending migrations
2024-03-03 13:27:18 No pending migrations
2024-03-03 13:27:18 2024-03-03T18:27:18.127946Z  INFO carp: Starting Carp
2024-03-03 13:27:18 2024-03-03T18:27:18.128009Z  INFO carp: Execution plan execution_plans/default.toml
2024-03-03 13:27:18 2024-03-03T18:27:18.128140Z  INFO carp: Config file PathBuf { inner: "/app/config/indexer/oura_docker.mainnet.yml" }
2024-03-03 13:27:18 2024-03-03T18:27:18.134168Z  INFO carp::sources::oura_source: Starting sync at block #169043 (869b30014807b9b158c8d24fb406deea93f39893169dc47c32a2cb6e6f642e2d)
2024-03-03 13:27:18 2024-03-03T18:27:18.134187Z  INFO carp::sources::oura_source: Attempting to connect to node...
2024-03-03 13:27:18 2024-03-03T18:27:18.134434Z  INFO carp::sources::oura_source: Connection to node established
2024-03-03 13:27:18 2024-03-03T18:27:18.134648Z  INFO carp::engine: Starting to process blocks
2024-03-03 13:27:18 2024-03-03T18:27:18.136230Z ERROR oura::sources::n2c::run: N2C error: Can't find chain intersection point    
2024-03-03 13:27:18 2024-03-03T18:27:18.136245Z  WARN oura::sources::n2c::run: unrecoverable error performing chainsync, will exit    
2024-03-03 13:27:18 2024-03-03T18:27:18.136447Z ERROR carp: Processing loop finished with error, stopping engine error=Can't fetch oura event: RecvError
2024-03-03 13:27:18 2024-03-03T18:27:18.136491Z  INFO carp: Engine is stopped successfully
```
It crashes I assume because the block hash doesn't match what's in the Cardano Node?

Any ideas what might be going on here?

**Comment by biofreeze** [comment]
Hey @rooooooooob is this now resolved? Should I close this issue?

**Comment by rooooooooob** [comment]
It's resolved in CML. I don't think carp has bumped the dependency version yet, it looks like it's still on CML 5.0.0. The fix is in 5.2.0. I believe that means that it *might* be fixed as "5.0.0" can use anything from that onwards but before 6.0.0, but it would depend which version is installed and used by cargo. I might be misunderstanding cargo's versioning behavior though. We should bump it in carp too. @ecioppettini
